### PR TITLE
Configurable claims for token validation

### PR DESCRIPTION
**Pull request by ybelMekk**
Check issue https://github.com/nais/tokendings/issues/25 for thoughts abut this PR.

dep: MockOuath2Server form 0.1.35 to 0.2.0
added optional claim as a field in TokenIssuer.kt
Update of mockServer needed som small changes as for audience as list.
Expanded TokenValidator.kt constructor
Updated test accordingly.

### CODE REVIEW

**Comment by ybelMekk** [comment]
Something with GH and the docker container? cant see it fail local.

**Comment by ybelMekk** [comment]
seems to happen when i run individual tests: `org.postgresql.util.PSQLException: ERROR: relation "rotatable_keys" does not exist?`
hmm

**Comment by jksolbakken** [comment]
Tests that use `withMockOAuth2Server()` expect a migrated database, so the migration should be run there. I did this in #20

**Review by jksolbakken — APPROVED** [review]
Bad naming a couple of places, looks good otherwise

File: src/main/kotlin/io/nais/security/oauth2/token/TokenValidator.kt [hunk 1]
```diff
@@ -9,13 +9,17 @@ import com.nimbusds.jwt.JWTClaimsSet
 import com.nimbusds.jwt.SignedJWT
 import java.net.URL
 
-class TokenValidator(private val issuer: String, jwkSource: JWKSource<SecurityContext?>) {
+class TokenValidator(
+    private val issuer: String,
+    jwkSource: JWKSource<SecurityContext?>,
+    private val optionalClaims: List<String>
```
[the remark this answers is not in the document]
Comment (ybelMekk) [reply src/main/kotlin/io/nais/security/oauth2/token/TokenValidator.kt:15]: confusing yeas.

**Comment by tronghn** [comment]
Solved in a761cbc0c3097d1abb7248f584851f7d10e3d0ab.

**Pull request closed without merging**